benchmarks/tensorexpr/__main__.py

```diff
@@ -133,13 +163,59 @@ def set_global_threads(num_threads):
 
     modes = args.mode.split(",")
 
+    datatypes = args.dtype.split(",")
+    for index, dtype in enumerate(datatypes):
+        datatypes[index] = getattr(torch, dtype)
+        if not datatypes[index] :
+            raise AttributeError("DataType: {} is not valid!".format(dtype))
+
     tensor_engine.set_engine_mode(args.engine)
 
     def run_default_configs(bench_cls, allow_skip=True):
-        for mode, device, config in itertools.product(
-            modes, devices, bench_cls.default_configs()
+        for mode, device, dtype, config in itertools.product(
+            modes, devices, datatypes, bench_cls.default_configs()
+        ):
+            bench = bench_cls(mode, device, dtype, *config)
+            bench.output_type = args.output
+            bench.jit_mode = args.jit_mode
+            if not bench.is_supported():
+                if allow_skip:
+                    continue
+                else:
+                    raise ValueError(
+                        "attempted to run an unsupported benchmark: %s" % (bench.desc())
+                    )
+            bench.run(args)
+
+    def run_with_input_iter(bench_cls, input_iter, allow_skip=True):
+        tensor_dim_specs = input_iter.split(',')
+        tensor_dim_specs = [dim.split(':') for dim in tensor_dim_specs]
+
+        configs = []
+        for dim in tensor_dim_specs :
+            assert len(dim) == 3, "There should be three dimensions for: start, stop, inc!"
```

Comment:
minor, this block of code would be clearer if `dim` were unpacked as `start, stop, inc = dim`

Reply:
I like your way better.  Thanks for the suggestion!